File: 2023/01.16/정호조/A와B.java
package baekjoon;

import java.util.*;
import java.io.*;
public class A와B_12904 {

	public static void main(String[] args) throws IOException{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		String s = br.readLine();
		String t = br.readLine();
		
		int ans = 0;
		
		while(t.length() > s.length()) {
			StringBuilder sb = new StringBuilder();

			if(t.equals(s)) {
				ans = 1;
				break;
			}
			if(t.charAt(t.length()-1) == 'A') {
				t = t.substring(0, t.length()-1);
			} else if(t.charAt(t.length()-1) == 'B') {
				t = t.substring(0,t.length()-1);
				t = sb.append(t).reverse().toString();
			}
		}
		
		System.out.println(ans);
	}
}

File: 2023/01.16/조수연/A와 B.kt
package com.ssafy.algorithm

import java.io.BufferedReader
import java.io.InputStreamReader

fun main() {

    val br = BufferedReader(InputStreamReader(System.`in`))

    var S = br.readLine()
    var T = br.readLine()

    var answer = 0

    for (i in T.indices) {
        val c = T.last()
        T = T.substring(0,T.length-1)

        // c가 B인 경우에는 자른 문자열을 뒤집어준다.
        if(c == 'B'){
            val sb = StringBuilder(T)
            T = sb.reverse().toString()
        }

        //두 문자열 비교
        if (S.length == T.length){
            if (S.equals(T)) answer = 1
            break
        }
    }

    println(answer)

}







